### PR TITLE
XDoc auto-generated doc

### PR DESCRIPTION
**Pull request by Kukks**
This PR introduces automatically generated assembly docs using XDoc and is built automatically via CI( AppVeyor) and then published to the gh_pages branch and makes it available under https://MetacoSA.github.io/NBitcoin/ 

You will need to replace the following in appveyor.yml  (use https://ci.appveyor.com/tools/encrypt to encrypt the values) 

github_access_token needs to be a a github access token with the public_repo permission.

```
  github_access_token:
    secure: /FamLw/uvGi0NNcmbw7tG8mNWgGK/bXTy6bS+KtaUb5Ztc9Qgy0Oxiz5BgBKzB4m   
  github_email:
    secure: p5VMsRKubRBjE6cqWpaDhNKEJPh0Z6Gddx2K//zAfus=
  github_user: Kukks
```
and the following in `docs/release.sh`
`git clone https://github.com/Kukks/NBitcoin.git --branch gh-pages $TEMP_REPO_DIR`
where you replace the url to https://github.com/MetacoSA/NBitcoin/
And then create an orphan branch named `gh_pages`

Demo here: https://kukks.github.io/NBitcoin/

### CODE REVIEW

File: docs/release.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+TEMP_REPO_DIR=$PWD/../NBitcoin-gh-pages
+docfx ./docs/docfx.json
+
+
+echo "Removing temporary doc directory $TEMP_REPO_DIR"
+rm -rf $TEMP_REPO_DIR
+mkdir $TEMP_REPO_DIR
+
+echo "Cloning the repo with the gh-pages branch"
+git clone https://github.com/Kukks/NBitcoin.git --branch gh-pages $TEMP_REPO_DIR
```
Comment (NicolasDorier) [inline docs/release.sh]: You might want to change that to MetacoSA?

**Comment by NicolasDorier** [comment]
this is so cool, thanks a lot for doing this!

**Comment by NicolasDorier** [comment]
Ignore my comments you want me to adapt to NBitcoin info right?

**Comment by Kukks** [comment]
Let me know when you want to give this a shot @NicolasDorier

**Comment by Kukks** [comment]
Let me know when you want to give this a shot @NicolasDorier

**Comment by NicolasDorier** [comment]
Oh sorry, I completely forgot about it, I will do it tomorrow. If I still forget let me know.

**Pull request merged**